contracts/mainnet/connectors/morpho-aave/main.sol

```diff
@@ -0,0 +1,442 @@
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		bool _isETH = _tokenAddress == ethAddr;
+
+		morphoAave.borrow(_poolTokenAddress, _amt);
+
+		if (_isETH) convertWethToEth(_isETH, TokenInterface(wethAddr), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogBorrow(address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_amt,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Borrow ETH/ERC20_Token with max gas.
+	 * @notice Borrow a token from Morpho Aave with max gas.
+	 * @param _tokenAddress The address of underlying token to borrow.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE).
+	 * @param _poolTokenAddress The address of aToken to borrow.(For ETH: aWETH address).
+	 * @param _amount The amount of the token (in underlying) to borrow.
+	 * @param _maxGasForMatching The maximum amount of gas to consume within a matching engine loop.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens borrowed.
+	 */
+	function borrowWithMaxGas(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _maxGasForMatching,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		bool _isETH = _tokenAddress == ethAddr;
+
+		morphoAave.borrow(_poolTokenAddress, _amt, _maxGasForMatching);
+
+		if (_isETH) convertWethToEth(_isETH, TokenInterface(wethAddr), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogBorrowWithMaxGas(address,address,uint256,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_amt,
+			_maxGasForMatching,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Withdraw ETH/ERC20_Token.
+	 * @notice Withdraw a token from Morpho Aave.
+	 * @param _tokenAddress The address of underlying token to withdraw.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of aToken to withdraw.(For ETH: aWETH address)
+	 * @param _amount The amount of the token (in underlying) to withdraw. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens withdrawed.
+	 */
+	function withdraw(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+		bool _isETH = _tokenAddress == ethAddr;
+		address _token = _isETH ? wethAddr : _tokenAddress;
+
+		if (_amt == uint256(-1))
+			(, , , _amt) = morphoAaveLens._getCurrentSupplyBalanceInOf(
+				_poolTokenAddress,
+				address(this)
+			);
+
+		morphoAave.withdraw(_poolTokenAddress, _amt);
+
+		convertWethToEth(_isETH, TokenInterface(_token), _amt);
```

Comment:
@MerlinEgalite, agree with this. But the thing is, we created Basic.sol helper library almost two years back and the same code is getting used everywhere. 
Updating it now would cause issues in other connectors. Also, the additional check before calling the function was for more optimization so that we don't call the function unnecessarily in case `_isETH` is `false`. Agree that we also have a check inside it.

Reply:
I understand, no problem!